# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/nsFontFaceLoader.h
# Commit: 43764851d067
# Full Hash: 43764851d067746870a65001786502244d7cf896
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-04-07 03:19:44
# Regressor Bug: 1694123
# File Overlap Count: 1
# Description:
#   Bug 1694123 - Pass userfont source index to font loader and decoder tasks, to avoid potential race with the field in the gfxUserFontEntry being reset on the main thread. r=layout-reviewers,emilio
#   
#   This is a preliminary step that is needed in order for the following patch to be safe. (It's
#   mostly just plumbing, although the patch looks big because we need to pass the index through
#   so many functions.)
# ==============================================================================

diff -r 9b111c26978a -r 43764851d067 layout/style/nsFontFaceLoader.h
--- a/layout/style/nsFontFaceLoader.h	Tue Apr 06 16:20:47 2021 +0000
+++ b/layout/style/nsFontFaceLoader.h	Tue Apr 06 16:23:17 2021 +0000
@@ -27,7 +27,7 @@
                                public nsIRequestObserver,
                                public nsIFontLoadCompleteCallback {
  public:
-  nsFontFaceLoader(gfxUserFontEntry* aFontToLoad, nsIURI* aFontURI,
+  nsFontFaceLoader(gfxUserFontEntry* aUserFontEntry, uint32_t aSrcIndex,
                    mozilla::dom::FontFaceSet* aFontFaceSet,
                    nsIChannel* aChannel);
 
@@ -65,6 +65,7 @@
   nsCOMPtr<nsITimer> mLoadTimer;
   mozilla::TimeStamp mStartTime;
   nsIStreamLoader* mStreamLoader;
+  uint32_t mSrcIndex;
   bool mInStreamComplete = false;
   bool mInLoadTimerCallback = false;
 };
